subprojects/launcher/src/main/java/org/gradle/launcher/cli/converter/LayoutToPropertiesConverter.java

```diff
@@ -42,8 +43,10 @@
 public class LayoutToPropertiesConverter {
 
     private final List<BuildOption<?>> allBuildOptions = new ArrayList<BuildOption<?>>();
+    private final ScriptFileResolver scriptFileResolver;
 
-    public LayoutToPropertiesConverter() {
+    public LayoutToPropertiesConverter(ScriptFileResolver scriptFileResolver) {
```

Comment:
❌  `ScriptFileResolver` is only a transitive dependency here. What `LayoutToPropertiesConverter` really needs is a way to produce a `BuildLayoutFactory` should it be needed.

Reply:
Right, only a `BuildLayoutFactory` is need here. It is always needed though, so no need for a "provider".

Reply:
Done